src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/multiblocks/ArmorForge.java

```diff
@@ -90,8 +91,16 @@ private void craft(Player p, ItemStack output, Inventory inv, Block dispenser) {
                     if (current < 3) {
                         p.getWorld().playSound(p.getLocation(), Sound.BLOCK_ANVIL_USE, 1F, 2F);
                     } else {
+                        Inventory outputInv2 = findOutputInventory(output, dispenser.getBlock(), dispenser.getInventory());
                         p.getWorld().playSound(p.getLocation(), Sound.ENTITY_ARROW_HIT_PLAYER, 1F, 1F);
-                        outputInv.addItem(output);
+                        if (outputInv2 != null) {
+                            outputInv2.addItem(output);
+                        } else if (InvUtils.fits(dispenser.getInventory(), output)) {
```

Comment:
You should cache these getInventory calls, it will make a new one every time.

(ref: https://hub.spigotmc.org/stash/projects/SPIGOT/repos/craftbukkit/browse/src/main/java/org/bukkit/craftbukkit/block/CraftDispenser.java#30-37)

Reply:
okay